### ISSUE TITLE
Face Recognition Question about Deep Learning Model

### ISSUE THREAD

**Issue by Aspirinkb** [davisking/dlib#847]
Thank you @davisking for your great work.

I'm using Dlib's face recognition interface and it works well. I know @davisking  trained the `ResNet-34` model as wrote in this blog '[High Quality Face Recognition with Deep Metric Learning](http://blog.dlib.net/2017/02/high-quality-face-recognition-with-deep.html)', and my questions are:
1. Is only there one ResNet model (a 29-layers CNN model) in this face recognition model ? Are there some other neural network layers or models combined?
2. Which layers in ResNet-32 remain?
3. Which loss function used when training?
4. Any training details?

**Comment by davisking** [comment]
All these questions are discussed in detail in the example programs.  Read http://dlib.net/dnn_face_recognition_ex.cpp.html
